heroic-elasticsearch-utils/src/main/java/com/spotify/heroic/elasticsearch/TransportClientSetup.java

```diff
@@ -51,10 +52,12 @@ public TransportClientSetup(
 
     @Override
     public Client setup() throws Exception {
-        final Settings settings =
-            ImmutableSettings.builder().put("cluster.name", clusterName).build();
+        final Settings settings = Settings.builder().put("cluster.name", clusterName).build();
 
-        final TransportClient client = new TransportClient(settings);
+        final TransportClient client = TransportClient.builder()
+            .settings(settings)
+            .addPlugin(DeleteByQueryPlugin.class)
```

Comment:
I'm on the fringe about the use of this plugin. Will the client work (apart from the delete-by-query usage) if its not installed server side?

Alternatively, implement a query-then-delete functionality on the client which seems to be the recommended approach.